### PR TITLE
MP-5134 Table Feature Editable

### PR DESCRIPTION
**Pull request by danilmoro3**
Ticket: https://spryker.atlassian.net/browse/MP-5134

### CODE REVIEW

File: docs/marketplace/dev/front-end/table-design/table-features/table-feature-editable.md [hunk 1]
```diff
@@ -0,0 +1,220 @@
...
+  - `cancelButon` - an object with `title` and `icon` for the `Cancel button`.  
+  - `disableForCols` - an array with cell `id's` to be disabled.  
+  - `formInputName` - will create `input[type=hidden]` element with the specific name.  
+  - `initialData` - initials data for cells and object with errors for rows and cells.  
+  
+`update`:
+  - `url` - request url.  
+  - `saveButon` - an object with `title` and `icon` for the `Save button` (displays in the `update` popup).  
+  - `cancelButon` - an object with `title` and `icon` for the `Cancel button` (displays in the `update` popup).  
+  - `disableForCols` - an array with cell `id's` to be disabled.  `
```
Comment (FRITS-Kh) [inline docs/marketplace/dev/front-end/table-design/table-features/table-feature-editable.md:54]:
```suggestion
  - `disableForCols` - an array with cell `ids` to be disabled.  `
```

File: docs/marketplace/dev/front-end/table-design/table-features/table-feature-editable.md [hunk 2]
```diff
@@ -0,0 +1,220 @@
...
+export type TableEditableConfigUrl = string | TableEditableConfigUrlObject;
+
+export interface TableEditableConfigButtonIcon {
+  icon: string;
+}
+
+export interface TableEditableConfigButtonText
+  extends Partial<TableEditableConfigButtonIcon> {
+  title: string;
+}
```
Comment (FRITS-Kh) [inline docs/marketplace/dev/front-end/table-design/table-features/table-feature-editable.md:189]:
```suggestion
  title: string;
  size: ButtonSize;
  shape: ButtonShape;
  variant: ButtonVariant;
  type: ButtonType;
}
```

**Review by FRITS-Kh — CHANGES_REQUESTED** [review]

File: docs/marketplace/dev/front-end/table-design/table-features/table-feature-editable.md [hunk 3]
```diff
@@ -0,0 +1,224 @@
...
+</spy-table>
+```
+
+Let's take a closer look at all the possible options.
+
+`columns`:  
+  - `id` - a cell `id`.  
+  - `type` - a cell `type`.  
+  - `typeOptions`:  
+    - `value` - will set default value to the cell of the newly added row.
```
Comment (supproduction) [inline docs/marketplace/dev/front-end/table-design/table-features/table-feature-editable.md:41]: Please, mention that you can use every `typeOptions` variant depended on table column type (with link to it)

File: docs/marketplace/dev/front-end/table-design/table-features/table-feature-editable.md [hunk 4]
```diff
@@ -0,0 +1,224 @@
...
+Let's take a closer look at all the possible options.
+
+`columns`:  
+  - `id` - a cell `id`.  
+  - `type` - a cell `type`.  
+  - `typeOptions`:  
+    - `value` - will set default value to the cell of the newly added row.  
+
+`create`:  
+  - `addButon` - an object with `title` and `icon` for the `Add button`.
```
Comment (supproduction) [inline docs/marketplace/dev/front-end/table-design/table-features/table-feature-editable.md:44]: It was updated recently, please fix it in this doc too

File: docs/marketplace/dev/front-end/table-design/table-features/table-feature-editable.md [hunk 5]
```diff
@@ -0,0 +1,224 @@
...
+})
+export class RootModule {}
+```
+
+## Interfaces
+
+Below you can find interfaces for Table Feature Editable.
+
+```ts
+export interface TableEditableColumn extends TableColumn {
```
Comment (supproduction) [inline docs/marketplace/dev/front-end/table-design/table-features/table-feature-editable.md:131]: Please, add interface to show that we extend default `TableConfig` with this feature

```
declare module '@spryker/table' {
  interface TableConfig {
    editable?: TableEditableConfig;
  }
}
```

File: docs/marketplace/dev/front-end/table-design/table-features/table-feature-editable.md [hunk 6]
```diff
@@ -0,0 +1,224 @@
...
+```ts
+@NgModule({
+  imports: [
+    TableModule.forRoot(),
+    TableModule.withFeatures({
+      editable: () => 
+        import('@spryker/table.feature.editable').then(
+          (m) => m.TableEditableFeatureModule,
+        ),   
+    }),
```
Comment (supproduction) [inline docs/marketplace/dev/front-end/table-design/table-features/table-feature-editable.md:120]: Also show example without dynamic registration, just feature tag inside table

**Review by supproduction — CHANGES_REQUESTED** [review]

File: docs/marketplace/dev/front-end/table-design/table-features/table-feature-editable.md (hunk 5 above)
Comment (danilmoro3) [reply docs/marketplace/dev/front-end/table-design/table-features/table-feature-editable.md:131]: added to the tech debt https://spryker.atlassian.net/browse/MP-5886

File: docs/marketplace/dev/front-end/table-design/table-features/table-feature-editable.md (hunk 6 above)
Comment (danilmoro3) [reply docs/marketplace/dev/front-end/table-design/table-features/table-feature-editable.md:120]: added to the tech debt https://spryker.atlassian.net/browse/MP-5886

File: docs/marketplace/dev/front-end/table-design/table-features/table-feature-editable.md [hunk 7]
```diff
@@ -123,6 +132,13 @@ Let's take a closer look at all the possible options.
 export class RootModule {}
 ```
 
+```html
+// Via HTML
```
Comment (supproduction) [inline docs/marketplace/dev/front-end/table-design/table-features/table-feature-editable.md:128]: Mention, that for this approach we need add TableEditableFeatureModule

```
@NgModule({
  imports: [
    TableModule.forRoot(),
    TableEditableFeatureModule,
  ],
})
export class RootModule {}
```

**Review by supproduction — CHANGES_REQUESTED** [review]

**Review by supproduction — DISMISSED** [review]

File: docs/marketplace/dev/front-end/table-design/table-features/table-feature-editable.md [hunk 8]
```diff
@@ -0,0 +1,246 @@
...
+
+`columns`:  
+  - `id` - a cell `id`.  
+  - `type` - a cell `type`.  
+  - `typeOptions` (see more about [Column Type](/docs/marketplace/dev/front-end/table-design/table-column-types/) options):  
+    - `value` - will set default value to the cell of the newly added row.  
+
+`create`:  
+  - `addButon` - an object with the `Add button` configuration like `title`, `icon`, `size`, etc..  
+  - `cancelButon` - an object with `title` and `icon` for the `Cancel button`.
```
Comment (FRITS-Kh) [inline docs/marketplace/dev/front-end/table-design/table-features/table-feature-editable.md:45]:
```suggestion
  - `cancelButon` - an object with the `Cancel button` configuration like `title`, `icon`, `size`, etc.
```

File: docs/marketplace/dev/front-end/table-design/table-features/table-feature-editable.md [hunk 9]
```diff
@@ -0,0 +1,246 @@
...
+  - `addButon` - an object with the `Add button` configuration like `title`, `icon`, `size`, etc..  
+  - `cancelButon` - an object with `title` and `icon` for the `Cancel button`.  
+  - `disableForCols` - an array with cell `ids` to be disabled.  
+  - `formInputName` - will create `input[type=hidden]` element with the specific name.  
+  - `initialData` - initials data for cells and object with errors for rows and cells.  
+  
+`update`:
+  - `url` - request url.  
+  - `saveButon` - an object with `title` and `icon` for the `Save button` (displays in the `update` popup).  
+  - `cancelButon` - an object with `title` and `icon` for the `Cancel button` (displays in the `update` popup).
```
Comment (FRITS-Kh) [inline docs/marketplace/dev/front-end/table-design/table-features/table-feature-editable.md:53]:
```suggestion
  - `saveButon` - an object with the `Save button` configuration like `title`, `icon`, `size`, etc (displays in the `update` popup).  
  - `cancelButon` - an object with the `Cancel button` configuration like `title`, `icon`, `size`, etc (displays in the `update` popup).  
```

**Review by FRITS-Kh — CHANGES_REQUESTED** [review]

**Review by FRITS-Kh — APPROVED** [review]

**Review by gibmirmilch — APPROVED** [review]

**Pull request merged**